packages/controller/lib/setup/setupBackup.js

```diff
@@ -649,36 +649,38 @@ class BackupRestore {
 
     restoreAfterStop(restartOnFinish, callback) {
         // Open file
-        let data = fs.readFileSync(tmpDir + '/backup/backup.json', 'utf8');
+        let data = fs.readFileSync(`${tmpDir}/backup/backup.json`, 'utf8');
         const hostname = tools.getHostName();
         data = data.replace(/\$\$__hostname__\$\$/g, hostname);
-        fs.writeFileSync(tmpDir + '/backup/backup_.json', data);
+        fs.writeFileSync(`${tmpDir}/backup/backup_.json`, data);
         let restore;
         try {
             restore = JSON.parse(data);
         } catch (e) {
-            console.error('Cannot parse "' + tmpDir + '/backup/backup_.json": ' + e);
+            console.error(`Cannot parse "${tmpDir}/backup/backup_.json": ${e.message}`);
             if (callback) {
                 callback(31);
             }
         }
 
         // stop all adapters
-        console.log('host.' + hostname + ' Clear all objects and states...');
+        console.log(`host.${hostname} Clear all objects and states...`);
         this.cleanDatabase(false, () => {
-            console.log('host.' + hostname + ' done.');
+            console.log(`host.${hostname} done.`);
             // upload all data into DB
-            // restore ioBorker.json
-            if (restore.config) {
+            // restore iobroker.json
+            if (restore.config && restore.config.system) {
```

Comment:
Ok, in fcat "system" should be always there, but now - if system is missing - the file would not get written at all ... 
I think the "if system" should be an if inside the if restore.config AND also checking that hostname is set at all .. because if hostname empty in current file then should also not be needed to be persisted or?!

Reply:
It’s not the main point of this pr, just added the check because I access properties of system